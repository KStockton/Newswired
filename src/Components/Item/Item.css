.Item-wrapper {
  background-image: url('../../Assets/solo.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  margin: 0;
  position: fixed;
  width: 100vw;
  background-position-y: center;
}