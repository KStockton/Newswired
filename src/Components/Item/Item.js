import React from 'react';
import {Link} from 'react-router-dom'


const Item = ({bookImage, rank, author, id, description, title, weeksOnList, articleImage, content}) => {

  return (
    <div>
      <Link to={`/main`} className="Item-back">
        Back
      </Link>
      <article className="Item-wrapper" id={id}>
        <section className="Item-content-wrapper">
          <img className="Item-img" src={bookImage|| articleImage} alt="book"/>
          <p className="Item-author">{author}</p>
          <p className="Item-rank">Rank: {rank}</p>
          <p className="Item-weeks">{weeksOnList} weeks on list</p>
          <p className="Item-title">{title}</p>
          <p className="Item-description">{description}</p>
          <p>{content || null}</p>
        </section>
      </article >
    </div>
  )
} 
export default Item