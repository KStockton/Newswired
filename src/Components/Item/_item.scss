.Item-back {
  background-color: #fdfcfc;
  position: fixed;
  top: 23%;
  left: 14%;
  padding: 10px;
  text-decoration: none;
  border: solid 1px #8ebed0;
  border-radius: 5px;
  text-transform: uppercase;
  z-index: 2;
  color: black;
  width: 65px;

  &:hover {
    padding-left: 10px;
    padding-right: 10px;
    padding-top: 11px;
    background-image: -webkit-gradient(linear, left bottom, left top, color-stop(0, #2cffc9), color-stop(1, #ffffff));
    color: white;
    background: black;
  }
}

.Item-back::before{
  background-color: #ccd0d56b;
  content: "";
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  padding: 8px;
  left: -8px;
  top: -8px;
  color: black;
  border-radius: 5px;
  box-shadow: inset 0px 1px 1px #000000, 0px 1px 0px #ff9494;
}

.Item-fontawesome {
  font-size: 18px;
}

.Item-wrapper {
  background-image: url('../Assets/solo.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  margin: 0;
  position: fixed;
  width: 100vw;
  z-index: -1;
  background-position-y: center;
}



.Item-content-wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 617px;
  background-color: #171616a1;
  margin: 8rem 21rem;
}

.Item-author{
  color: white;
  padding: 8px;
  font-size: 13px;
  font-style: italic;
  text-align: left;
}
.Item-rank{
  color: white;
  padding: 8px;
}
.Item-weeks {
  color: white;
  padding: 8px;
}
.Item-title{
  color:white;
  padding: 8px;
  max-width: 544px;
  font-weight: 200;
  font-size: 21px;
  font-family: 'Old Standard TT', serif;
}
.Item-description{
  color: white;
  padding: 8px;
  max-width: 500px;
  text-align: left;
  padding-top: 20px;
  font-family: 'Old Standard TT', serif;
}

.Item-img {
  max-height: 300px;
  max-width: 500px;
  padding: 12px 0px;
}
.Item-content {
  color: white;
  max-width: 500px;
  text-align: left;
  font-family: 'Old Standard TT', serif;
}