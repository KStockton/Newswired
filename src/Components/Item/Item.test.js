import React from 'react';
import Item from './Item';
import { shallow } from 'enzyme';

describe('Item', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Item/>) 
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})