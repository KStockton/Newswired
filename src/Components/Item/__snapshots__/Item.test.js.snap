// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match the snapshot 1`] = `
<div>
  <Link
    className="Item-back"
    to="/main"
  >
    Back
  </Link>
  <article
    className="Item-wrapper"
  >
    <section
      className="Item-content-wrapper"
    >
      <img
        alt="book"
        className="Item-img"
      />
      <p
        className="Item-author"
      />
      <p
        className="Item-rank"
      >
        Rank: 
      </p>
      <p
        className="Item-weeks"
      >
         weeks on list
      </p>
      <p
        className="Item-title"
      />
      <p
        className="Item-description"
      />
      <p
        className="Item-content"
      />
    </section>
  </article>
</div>
`;
